src/servers/launcher/main.cpp

```diff
@@ -0,0 +1,849 @@
+#include <libpq-fe.h>
+
+#include <array>
+#include <cerrno>
+#include <chrono>
+#include <cstdint>
+#include <cstring>
+#include <expected>
+#include <iomanip>
+#include <iostream>
+#include <netinet/in.h>
+#include <ranges>
+#include <signal.h>
+#include <stdexcept>
+#include <sstream>
+#include <string>
+#include <sys/epoll.h>
+#include <sys/wait.h>
+#include <thread>
+#include <unistd.h>
+#include <utility>
+#include <variant>
+
+using namespace std::literals;
+using notManyMilliseconds = std::chrono::duration<int, std::milli>;
+using PacketBuffer = std::array<uint8_t, 65536>;
+
+static constexpr const auto RETRY_DELAY{1s};
+static constexpr const uint16_t PORT{32232};
+static constexpr const uint8_t CABOOSE{108};
+
+struct UUID {
+  std::array<const uint8_t, 16> uuid;
+
+  UUID() : uuid(std::array<const uint8_t, 16>{
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
+  }) {}
+  UUID(std::ranges::sized_range auto&& uuid) :
+    uuid([&uuid]{
+      static_assert(uuid.size() == 16, "UUIDs must be 16 bytes long");
+
+      return std::array<const uint8_t, 16>{
+        uuid[0],
+        uuid[1],
+        uuid[2],
+        uuid[3],
+        uuid[4],
+        uuid[5],
+        uuid[6],
+        uuid[7],
+        uuid[8],
+        uuid[9],
+        uuid[10],
+        uuid[11],
+        uuid[12],
+        uuid[13],
+        uuid[14],
+        uuid[15],
+      };
+    }()) {}
+  static UUID unsafeCreate(const char* v) {
```

Comment:
I would ditch the unsafe create operation completely; it just makes no sense. The user knows their data is large enough, so let them explicitly promise so:

```cpp
std::byte* data = // ... generate data somehow...

// instead of:
auto uuid = UUID::unsafeCreate(data);
// make the user do:
auto uuid = UUID{std::span{data, 16}};
// or:
auto uuid = UUID{std::subrange{data, data + 16}};
// or something like that
```

If they’re wrong about the size of the data, then make it their problem, not yours.